dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.h

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_GPU_H_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_GPU_H_
+
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/kernels/common/block_setup.h"
+
+namespace dali {
+namespace kernels {
+namespace brightness_contrast {
+
+
+template <size_t ndims>
+using Roi = Box<ndims, int>;
+
+
+template <class InputType, class OutputType, int ndims>
```

Comment:
I followed [Google style guidelines](https://google.github.io/styleguide/cppguide.html#Output_Parameters), which say to use input->output order. Which BTW I think it's more intuitive, first you need to provide input, to receive the output. And I don't think there's any place, which clearly says, we use output->input order

Reply:
I have no idea, why `memcpy` takes this precise order. Input->Output order is more intuitive for me and I would use it. 
We don't keep such consistencies in our code (e.g. using `///` and `/** */` for documentation), so I don't feel obligated to keep it in this order simply because it was used before.